src/Middleware.php

```diff
@@ -3,31 +3,138 @@
+     *
+     * @see https://inertiajs.com/asset-versioning
+     * @param  \Illuminate\Http\Request  $request
+     * @return string|null
+     */
+    public function version(Request $request)
     {
-        $response = $next($request);
+        // When we are running on Laravel Vapor, asset URLs are automatically updated
+        // when the application is being deployed. Because of this, we do not need
+        // to use any files for hashing, as we can simply use this URL instead.
+        if (config('app.asset_url')) {
+            return md5(config('app.asset_url'));
+        }
 
-        if (!$request->header('X-Inertia')) {
-            return $response;
+        // Alternatively, when we are running in a regular (non-serverless) environment
+        // we'll attempt to use the Laravel Mix asset manifest to generate our hash.
+        if (file_exists($manifest = public_path('mix-manifest.json'))) {
+            return md5_file($manifest);
         }
+    }
+
+    /**
+     * Defines the props that are shared by default.
+     *
+     * @see https://inertiajs.com/shared-data
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function share(Request $request)
+    {
+        return [
+            'errors' => function () use ($request) {
+                return $this->resolveValidationErrors($request);
+            },
+        ];
+    }
+
+    /**
+     * Handle the incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  Closure  $next
+     * @return Response
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        Inertia::share($this->share($request));
+
+        Inertia::version(function () use ($request) {
+            return $this->version($request);
+        });
+
+        $response = $next($request);
+        $response = $this->checkVersion($request, $response);
+        $response = $this->changeRedirectCode($request, $response);
+
+        return $response;
+    }
 
-        if ($request->method() === 'GET' && $request->header('X-Inertia-Version', '') !== Inertia::getVersion()) {
+    /**
+     * In the event that the asset version changes, Inertia will automatically reload
+     * the page in order to ensure the application keeps working for the end user.
+     *
+     * @param  Request  $request
+     * @param  Response  $response
+     * @return Response
+     */
+    public function checkVersion(Request $request, Response $response)
+    {
+        if ($request->header('X-Inertia') &&
+            $request->method() === 'GET' &&
+            $request->header('X-Inertia-Version', '') !== Inertia::getVersion()
+        ) {
             if ($request->hasSession()) {
                 $request->session()->reflash();
             }
 
-            return Response::make('', 409, ['X-Inertia-Location' => $request->fullUrl()]);
+            return new Response('', 409, ['X-Inertia-Location' => $request->fullUrl()]);
```

Comment:
So obvious, but we can now do this:

```suggestion
            return Inertia::location($request->fullUrl());
```